appinventor/components/src/com/google/appinventor/components/runtime/util/JavaReplaceWithMappings.java

```diff
@@ -0,0 +1,189 @@
+// -*- mode: java; c-basic-offset: 2; -*-
```

Comment:
I would consider merging the logic of JavaJoinListOfStrings and JavaReplaceWithMappings into a single utility class, e.g., JavaStringUtils. I think it's better than introducing a number of very small, targeted classes that hold only one or two methods each.

Reply:
Agreed; Done!